cleanlab/token_classification/rank.py

```diff
@@ -0,0 +1,112 @@
+import pandas as pd
+import numpy as np
+from cleanlab.rank import get_label_quality_scores as main_get_label_quality_scores
+
+
+def get_label_quality_scores(
+    labels: list,
+    pred_probs: list,
+    *,
+    tokens: list = None,
+    token_scoring_method: str = "self_confidence",
+    sentence_scoring_method: str = "min",
+    param: float = 0.04,
+    return_scores_per_token: bool = True,
+):
+    """Returns label quality scores for each sentence
+
+    This is a function to compute label-quality scores for token classification datasets, where lower scores
+    indicate labels less likely to be correct.
+
+    Score is between 0 and 1.
+
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    If `return_scores_per_token` is set to True, also return label score per token
+
+    Parameters
+    ----------
+    labels: list
+        noisy token labels in nested list format, such that `labels[i]` is a list of token labels of the i'th
+        sentence. For datasets with `K` classes, each label must be in 0, 1, ..., K-1. All classes must be present.
+
+    pred_probs: list
+        list of np.arrays, such that `pred_probs[i]` is the model-predicted probabilities for the tokens in
+        the i'th sentence, and has shape `(N, K)`. Each row of the matrix corresponds to a token `t` and contains
+        the model-predicted probabilities that `t` belongs to each possible class, for each of the K classes. The
+        columns must be ordered such that the probabilities correspond to class 0, 1, ..., K-1.
+
+    sentence_scoring_method: {"min", "softmin"}, default="min"
+        sentence scoring method to aggregate token scores.
+
+        - `min`: sentence score = minimum token label score of the sentence
+        - `softmin`: sentence score = <s, softmax(1-s, t)>, where s denotes the token label scores of the sentence,
+        and <a, b> == np.dot(a, b). The parameter `t` controls parameter of softmax, such that when t -> 0, the
+        method approaches to `min`. This method is the "softer" version of `min`, which adds some minor weights to
+        other scores.
+
+    token_scoring_method: {"self_confidence", "normalized_margin", "confidence_weighted_entropy"}, default="self_confidence"
+        label quality scoring method. See `cleanlab.rank.get_label_quality_scores` for more info.
+
+    param: float, default=0.04
```

Comment:
Is 0.05 noticeably worse in our benchmarks?  If not, I'd advocate for that as the default value as it's a more standard number.

Reply:
Can we also change the name of this variable (later in the code since I suggested removing it from here) to be: `temperature` instead of `param`?